xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -866,42 +913,53 @@ void CAESinkAUDIOTRACK::UpdateAvailablePassthroughCapabilities()
     {
       if (CJNIAudioFormat::ENCODING_DTS_HD != -1)
       {
-        CLog::Log(LOGDEBUG, "Firmware implements DTS-HD RAW");
-        m_info.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTSHD);
+        if (VerifySinkConfiguration(48000, AEChannelMapToAUDIOTRACKChannelMask(AE_CH_LAYOUT_7_1), CJNIAudioFormat::ENCODING_DTS_HD))
+        {
+          CLog::Log(LOGDEBUG, "Firmware implements DTS-HD RAW");
+          m_info.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_DTSHD);
+        }
       }
       if (CJNIAudioFormat::ENCODING_DOLBY_TRUEHD != -1)
       {
-        CLog::Log(LOGDEBUG, "Firmware implements TrueHD RAW");
-        m_info.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_TRUEHD);
+        if (VerifySinkConfiguration(48000, AEChannelMapToAUDIOTRACKChannelMask(AE_CH_LAYOUT_7_1), CJNIAudioFormat::ENCODING_DOLBY_TRUEHD))
```

Comment:
Should introduce some constants for these magic numbers

Reply:
Which one is magic?

Reply:
48000 used here and on the other places and 192000 etc.

Reply:
Yeah. Those are the two default rates when doing DTS / AC3 passthrough or when doing DTS-HD, TRUEHD Passthrough. Are used this way all over the sinks. I don't really see the point in changing this as it's clear for every audio guy.

Reply:
Never the less. If it's very important for you - I can do that (at least for the enumeration code).

Reply:
It's your choice. For me an advantage for using constants is finding references instead of grepping the number.

Reply:
I'll keep as is, cause of the following: if you wake me up at night and asked me about the default DTS samplerate when doing Passthrough I would answerk 48k. if you'd ask me how I named the 48000 in AESinkAudioTrack I needed to think :-) - in fact I would search for 48000 to find the var name and from there I would search for the varName .